ethcore/private-tx/src/lib.rs

```diff
@@ -512,25 +525,36 @@ impl Provider where {
 			let (new_address, _) = ethcore_contract_address(engine.create_address_scheme(env_info.number), &sender, &nonce, &transaction.data);
 			Some(new_address)
 		});
+		let contract_address = contract_address.expect("Private contract address is non zero by this point, \
```

Comment:
Maybe change the above line to `unwrap_or_else(||` ? 
The proof won't be necessary and also the contract address will be evaluated lazily.